packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php

```diff
@@ -48,16 +49,28 @@
             </span>
 
             @if (filled($badge))
-                <span
-                    @class([
-                        'inline-flex items-center justify-center ml-auto rtl:ml-0 rtl:mr-auto min-h-4 px-2 py-0.5 text-xs font-medium tracking-tight rounded-xl whitespace-normal',
-                        'text-primary-700 bg-primary-500/20' => ! $active,
-                        'text-white bg-white/20' => $active,
-                        'dark:text-primary-500' => (! $active) && config('filament.dark_mode'),
-                    ])
-                >
-                    {{ $badge }}
-                </span>
+                <span @class([
+                    'inline-flex items-center justify-center ml-auto rtl:ml-0 rtl:mr-auto min-h-4 px-2 py-0.5 text-xs font-medium tracking-tight rounded-xl whitespace-normal',
+                    'text-white bg-white/20' => $active,
+                    ...match ($badgeColor ?? null) {
+                        'danger' => [
+                            'text-danger-700 bg-danger-500/10' => ! $active,
+                            'dark:text-danger-500' => (! $active) && config('tables.dark_mode'),
+                        ],
+                        'success' => [
+                            'text-success-700 bg-success-500/10' => ! $active,
+                            'dark:text-success-500' => (! $active) && config('tables.dark_mode'),
+                        ],
+                        'warning' => [
+                            'text-warning-700 bg-warning-500/10' => ! $active,
+                            'dark:text-warning-500' => (! $active) && config('filament.dark_mode'),
+                        ],
+                        default => [
+                            'text-primary-700 bg-primary-500/20' => ! $active,
+                            'dark:text-primary-500' => (! $active) && config('filament.dark_mode'),
+                        ],
+                    },
```

Comment:
Could this be cleaned up by extracting this into a method? Would also allow people to add custom colors.

Reply:
This was inherited from #3218

Reply:
We could use the `default` arm to allow custom colors

Reply:
Done

Reply:
Hmm, custom colors would need to be hard-coded for Tailwind CSS to compile them.

Reply:
Yes, the user passes in custom color classes and adds them to the tailwind config

Reply:
Sorry if I wasn't clear, but Tailwind needs the color classes to be in the Blade view. Otherwise they aren't compiled into the production CSS.

Reply:
I know, they need to add the resource to the tailwind.config content, so the classes from the file are added to the CSS.

Reply:
Okay, if that's the intended route, then it's fine.